File: MongoBooks2/AngularMongoBooks3/ClientApp/src/app/Models/tallyDelta.ts
import { Book } from './book'

export interface ITallyDelta
{
  daysInTally: number;
  totalPages: number;
  totalBooks: number;
  totalBookFormat: number;
  totalComicFormat: number;
  totalAudioFormat: number;
  percentageInEnglish: number;
  percentageInTranslation: number;
  pageRate: number;
  daysPerBook: number;
  pagesPerBook: number;
  booksPerYear: number;
};

export class TallyDelta implements ITallyDelta
{
  constructor(
    public daysInTally: number = 0,
    public totalPages: number = 0,
    public totalBooks: number = 0,
    public totalBookFormat: number = 0,
    public totalComicFormat: number = 0,
    public totalAudioFormat: number = 0,
    public percentageInEnglish: number = 0,
    public percentageInTranslation: number = 0,
    public pageRate: number = 0,
    public daysPerBook: number = 0,
    public pagesPerBook: number = 0,
    public booksPerYear: number = 0)
  {
  }
}

File: MongoBooks2/BooksControllerUtilities/DataClasses/TallyDelta.cs
namespace BooksControllerUtilities.DataClasses
{
    using BooksCore.Books;

    public class TallyDelta
    {
        public int DaysInTally { get; set; }

        public int TotalPages { get; set; }

        public int TotalBooks { get; set; }

        public int TotalBookFormat { get; set; }

        public int TotalComicFormat { get; set; }

        public int TotalAudioFormat { get; set; }

        public float PercentageInEnglish { get; set; }

        public float PercentageInTranslation { get; set; }

        public float PageRate { get; set; }

        public float DaysPerBook { get; set; }

        public float PagesPerBook { get; set; }

        public float BooksPerYear { get; set; }

        public TallyDelta()
        {

        }

        public TallyDelta(BooksDelta.DeltaTally talliedBook)
        {
            DaysInTally = (int)talliedBook.DaysInTally;
            TotalPages = (int)talliedBook.TotalPages;
            TotalBooks = (int)talliedBook.TotalBooks;
            TotalBookFormat = (int)talliedBook.TotalBookFormat;
            TotalComicFormat = (int)talliedBook.TotalComicFormat;
            TotalAudioFormat = (int)talliedBook.TotalAudioFormat;

            PercentageInEnglish = (float)talliedBook.PercentageInEnglish;
            PercentageInTranslation = (float)talliedBook.PercentageInTranslation;
            PageRate = (float)talliedBook.PageRate;
            DaysPerBook = (float)talliedBook.DaysPerBook;
            PagesPerBook = (float)talliedBook.PagesPerBook;
            BooksPerYear = (float)talliedBook.BooksPerYear;
        }

    }
}
